telegram/_shared.py

```diff
@@ -37,40 +48,89 @@ class UsersShared(TelegramObject):
+            bits, so 64-bit integers or double-precision float types are safe for storing these
+            identifiers. The bot may not have access to the users and could be unable to use
+            these identifiers, unless the users are already known to the bot by some other means.
+            Mutually exclusive with :paramref:`users`.
+
+            .. versionchanged:: NEXT.VERSION
+                Bot API 7.2 introduced :paramref:`users` replacing this argument. PTB will
+                automatically convert this to that one, but for advanced options, please use
+                :paramref:`users` directly.
+
+            .. deprecated:: NEXT.VERSION
+                In future versions, this argument will become keyword only.
 
     Attributes:
         request_id (:obj:`int`): Identifier of the request.
-        user_ids (Tuple[:obj:`int`]): Identifiers of the shared users. These numbers may have
-            more than 32 significant bits and some programming languages may have difficulty/silent
-            defects in interpreting them. But they have at most 52 significant bits, so 64-bit
-            integers or double-precision float types are safe for storing these identifiers. The
-            bot may not have access to the users and could be unable to use these identifiers,
-            unless the users are already known to the bot by some other means.
+        users (Tuple[:class:`telegram.SharedUser`]): Information about users shared with the
+            bot. Mutually exclusive with :attr:`user_ids`.
+
+            .. versionadded:: NEXT.VERSION
     """
 
-    __slots__ = ("request_id", "user_ids")
+    __slots__ = ("request_id", "users")
 
     def __init__(
         self,
         request_id: int,
-        user_ids: Sequence[int],
+        users: Sequence["SharedUser"],
+        user_ids: Optional[Sequence[int]] = None,
         *,
         api_kwargs: Optional[JSONDict] = None,
     ):
         super().__init__(api_kwargs=api_kwargs)
         self.request_id: int = request_id
-        self.user_ids: Tuple[int, ...] = tuple(user_ids)
+        self.users: Tuple[SharedUser, ...] = parse_sequence_arg(users)
+        self.user_ids: Optional[Tuple[int, ...]] = parse_sequence_arg(  # type: ignore[misc]
+            user_ids
+        )
+        if user_ids is not None and users:
+            raise ValueError("`users` and `user_ids` are mutually exclusive")
+
+        if user_ids is not None:
+            warn(
+                build_deprecation_warning_message(
+                    deprecated_name="user_ids",
+                    new_name="users",
+                    object_type="parameter",
+                    bot_api_version="7.2",
+                ),
+                PTBDeprecationWarning,
+                stacklevel=2,
+            )
 
-        self._id_attrs = (self.request_id, self.user_ids)
+        self._id_attrs = (self.request_id, self.users)
```

Comment:
please update that in the class docstring & add a versionchanged to the class as well :)